index.html

```diff
@@ -2175,6 +2297,84 @@ <h3>Verify Proof Sets and Chains</h3>
         </ol>
       </section>
 
+      <section class="normative">
+        <h3>Context Validation</h3>
+
+        <p>
+The following algorithm provides one mechanism that can be used to ensure that
+an application understands the contexts associated with a document before it
+executed business rules specific to the input in the document. For more
+rationale related to this algorithm, see Section [[[#validating-contexts]]].
+This algorithm takes inputs of a document ([=map=] |inputDocument|), a set of
+approved JSON-LD Contexts ([=map=] |expectedContext|), and a boolean to
+recompact when unknown contexts are detected ([=boolean=] |recompact|), and
+returns a [=map=] that contains the following:
+        </p>
+
+        <ul>
+          <li>
+a status ([=boolean=] |status|)
+          </li>
+          <li>
+a validated document ([=map=] |document|) if no errors occurred
+          </li>
+          <li>
+zero or more warnings ([=list=] of [=ProblemDetails=] |warnings|)
+          </li>
+          <li>
+zero or more errors ([=list=] of [=ProblemDetails=] |errors|)
+          </li>
+        </ul>
+
+        <p>
+The context validation algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set |result|.|status| to `false`, |result|.|warnings| to an empty list,
+|result|.|errors| to an empty list, |compactionContext| to an empty list;
+and clone |inputDocument| to |result|.|document|.
+          </li>
+          <li>
+Let |contextValue| be the value of the `@context` property of |result|.|document|,
+which might be undefined.
+          </li>
+          <li>
+If |contextValue| does not deeply equal |expectedContext|, then if any subtree in
```

Comment:
Slight mixup here in the grammar fixes, this step should read "or if" not "then if".